ethcore/src/miner/miner.rs

```diff
@@ -503,12 +526,11 @@ impl Miner {
 		};
 
 		let block_start = Instant::now();
-		debug!(target: "miner", "Attempting to push {} transactions.", pending.len());
+		debug!(target: "miner", "Attempting to push {} transactions.", engine_pending.len() + queue_pending.len());
 
-		for tx in pending {
+		for transaction in engine_pending.into_iter().chain(queue_pending.into_iter().map(|tx| tx.signed().clone())) {
```

Comment:
Here there's a potential problem I think: if the engine's txs can't **all** be included in the block we should…  do what? wait for the next block? 
Either way: as it stands, unless all the engine's txs are included there's a problem right? Maybe add them separately (instead of `.chain()`-ing them with the normal txs)?

Reply:
I'm still hesitating:
If we merge #11245, this will be used for queued malice reports. If those don't all fit into a block, that's not catastrophic at all: They'd just be retried. But if we return an error here, we'd be unable to ever produce a block again in that scenario.

We could instead allow `on_prepare_block` to mark its returned transactions as critical. Then block creation would only fail if a _critical_ transaction failed to be included.